cypress/integration/tests/administrator/repository/git.test.ts

```diff
@@ -0,0 +1,126 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+/// <reference types="cypress" />
+
+import {
+    deleteAllBusinessServices,
+    deleteApplicationTableRows,
+    getRandomAnalysisData,
+    getRandomApplicationData,
+    hasToBeSkipped,
+    login,
+    preservecookies,
+    resetURL,
+    writeMavenSettingsFile,
+} from "../../../../utils/utils";
+import * as data from "../../../../utils/data_utils";
+import { GitConfiguration } from "../../../models/administrator/repositories/git";
+import { Analysis } from "../../../models/developer/applicationinventory/analysis";
+import { CredentialsSourceControlUsername } from "../../../models/administrator/credentials/credentialsSourceControlUsername";
+import { CredentialType, UserCredentials } from "../../../types/constants";
+
+let gitConfiguration = new GitConfiguration();
+let source_credential;
+
+describe("Test an application form a Git source", { tags: "@tier1" }, () => {
+    before("Login", function () {
+        // Prevent hook from running, if the tag is excluded from run
+        if (hasToBeSkipped("@tier1")) return;
+
+        login();
+        deleteApplicationTableRows();
+        source_credential = new CredentialsSourceControlUsername(
+            data.getRandomCredentialsData(
+                CredentialType.sourceControl,
+                UserCredentials.usernamePassword,
+                true
+            )
+        );
+        source_credential.create();
+    });
+
+    beforeEach("Persist session", function () {
+        // Save the session and token cookie for maintaining one login session
+        preservecookies();
+        cy.fixture("application").then(function (appData) {
+            this.appData = appData;
+        });
+        cy.fixture("analysis").then(function (analysisData) {
+            this.analysisData = analysisData;
+        });
+
+        // Interceptors
+        cy.intercept("POST", "/hub/application*").as("postApplication");
+        cy.intercept("GET", "/hub/application*").as("getApplication");
+    });
+
+    afterEach("Persist session", function () {
+        // Reset URL from report page to web UI
+        resetURL();
+    });
+
+    after("Perform test data clean up", () => {
+        if (hasToBeSkipped("@tier1")) return;
+        deleteApplicationTableRows();
+        deleteAllBusinessServices();
+        source_credential.delete();
+        writeMavenSettingsFile(data.getRandomWord(5), data.getRandomWord(5));
+    });
+
+    it("Enable Insecure git Repository", () => {
+        GitConfiguration.open();
+        gitConfiguration.toggleInsecureGitRepositories();
+    });
+
+    it("Source code analysis on tackle testapp", function () {
```

Comment:
it("Analysis on secure(https) git repo for tackle test app", function(){
        
       
        GitConfiguration.open();
        gitConfiguration.toggleInsecureGitRepositories();
        const application = new Analysis(       getRandomApplicationData("Secure_enabled_tackle_test_app", {
                sourceData: this.appData[3],
            }),
})